inprocess/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -786,6 +790,28 @@ public void writeMessage(InputStream message) {
           statsTraceCtx.outboundMessageSent(outboundSeqNo, -1, -1);
           serverStream.statsTraceCtx.inboundMessage(outboundSeqNo);
           serverStream.statsTraceCtx.inboundMessageRead(outboundSeqNo, -1, -1);
+
+          long messageLength = -1;
+          try {
+            messageLength = getKnownLength(message);
+          } catch (IOException e) {
+            throw Status.INTERNAL
+                    .withDescription("Failed to calculate size")
+                    .withCause(e)
+                    .asRuntimeException();
+          } catch (StatusRuntimeException e) {
+            throw e;
+          } catch (RuntimeException e) {
+            throw Status.INTERNAL
+                    .withDescription("Failed to calculate size")
+                    .withCause(e)
+                    .asRuntimeException();
+          }
+          statsTraceCtx.outboundUncompressedSize(messageLength);
+          statsTraceCtx.outboundWireSize(messageLength);
+          // messageLength should be same at receiver's end as no actual wire is involved.
+          serverStream.statsTraceCtx.inboundUncompressedSize(messageLength);
```

Comment:
Given that you are reporting on the server side, why not just get rid of the if statements around sizes reported?

Reply:
Updated with the new commit by reporting sizes on both client side and server sides and removing if statements. 

Correction: In previous commit sizes were reported only on client side.